Last(?) Updates to dual-funding Spec

niftynei opened the pull request:
Adds requested spec proposal changes, including addition of `est_witness_weight` field and `second_commitment_point` to the open/accept_channel2 messages.

(Note that `accept_channel2` also includes `second_commitment_point`, per commit https://github.com/lightning/bolts/pull/851/commits/ee5bef78aba209ad9f28fbf1f40fb432fea6d108).

Built on top of #5900.

@t-bast this is a good one to test for interop!

rustyrussell reviewed (changes requested):
Ack ef98b1042efd8869b4e34df68e20902905435a48

Please put something for CHANGELOG.md?

`Changelog-EXPERIMENTAL: Protocol: `dual-funding` spec changed in incompatible ways, won't work with old versions (but maybe soon with Eclair!)`

t-bast commented:
> including addition of est_witness_weight field

But on the contrary, we decide that we should drop that field, not add it?

The changes that we discussed during the last spec meeting that are still missing from the spec are:

- adding `second_per_commitment_point` to `accept_channel2`
- removing feature dependency on `static_remotekey`
- removing `est_witness_weight`
- adding a rationale paragraph on the fee griefing attack and the need to actively double-spent inputs in concurrent interactive-tx attempts

At first glance, this PR implements the first point but not the other two?

niftynei commented:
Thanks for the review @rustyrussell + @t-bast.

Ah ok, I'm on board with removing `est_witness_weight`. Updated here and in the spec. We still verify that the provided bytes at least meet/match the provided feerate.

Also removed the assumption that `static remotekey` is flagged on.

I think that's everything?? Will keep an eye on the CI.

rustyrussell commented:
Ack 91b04af3527446ac51bd557c6e2f023336b15f41

rustyrussell commented:
Trivial rebase on master, now #5900 is merged, as GH complained about conflicts for some reason.

Ack c7d8c297f6c9a6358372d456a5ab4ec1dbcf178c

t-bast commented:
Awesome, thanks @niftynei I'll start working on cross-compatibility tests!
